llvm/include/llvm/Support/TimeProfiler.h

```diff
@@ -83,16 +83,27 @@ namespace llvm {
 
 class raw_pwrite_stream;
 
+struct TimeTraceMetadata {
+  std::string Detail;
+  // Source file information for the event.
+  std::string Filename;
```

Comment:
NIT: Maybe use `File` here and `file` in JSON for brevity? 
I'm not too worried about the code, but it looks like the JSON trace format does attempt to be brief, and having a shorter name should be aligned with that (and will make the file slightly smaller, although not by a huge margin)

Reply:
Done.